src/Stream.php

```diff
@@ -61,6 +61,7 @@ private function __construct()
      * @param string|resource|StreamInterface $body
      *
      * @throws \InvalidArgumentException
+     * @throws \ValueError
```

Comment:
We should have a consistent API across all PHP versions. The internals of fopen changing should not effect the public API of this class.

Reply:
This can be removed, right?

Reply:
I guess we can actually make it so there are no extra `fopen`s left at all with something like:

```diff
@@ -69,12 +69,6 @@ class Stream implements StreamInterface
             return $body;
         }
 
-        if (\is_string($body)) {
-            $resource = \fopen('php://temp', 'rw+');
-            \fwrite($resource, $body);
-            $body = $resource;
-        }
-
         if (\is_resource($body)) {
             $new = new self();
             $new->stream = $body;
@@ -86,6 +80,13 @@ class Stream implements StreamInterface
             return $new;
         }
 
+        if (\is_string($body)) {
+            $stream = self::createFromFile('php://temp', 'rw+');
+            $stream->write($body);
+
+            return $stream;
+        }
+
         throw new \InvalidArgumentException('First argument to Stream::create() must be a string, resource or StreamInterface.');
     }
 ```